packages/react/src/components/DataTable/TableHeader.tsx

```diff
@@ -91,7 +158,7 @@ const TableHeader = React.forwardRef(function TableHeader(
     [`${prefix}--table-sort--descending`]:
       isSortHeader && sortDirection === sortStates.DESC,
   });
-  const ariaSort = !isSortHeader ? 'none' : sortDirections[sortDirection];
+  const ariaSort = !isSortHeader ? 'none' : sortDirection ? sortDirections[sortDirection] : sortDirections[sortStates.NONE];
```

Comment:
It is still failing here. I tried debugging but got no luck. Any help or suggestion is appreciated. 😃